### PR TITLE
Shuffle

### PR DESCRIPTION
**Pull request by railsagainstignorance**
shuffles the articleId before adding to/subtracting from the sharecode,
also some clarification of the zero padding code

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -201,5 +244,10 @@ function isShareCodePattern(code) {
 module.exports = {
 	encrypt: encrypt,
 	decrypt: decrypt,
-	isShareCodePattern: isShareCodePattern
+	isShareCodePattern: isShareCodePattern,
+	_removeHyphens: removeHyphens,
```
Comment (JakeChampion) [inline index.js:246]: Why export these?

File: index.js [hunk 2]
```diff
@@ -126,11 +132,30 @@ function validateStringOrThrow(name, value, regexp){
 	}
 }
 
+// any -ve num is coerced to -1
+
+function constructMaxTokensString( num, maxLength ) {
+	if (maxLength < 2) {
+		throw new Error( "constructMaxTokensString: maxLength(", maxLength, ") is too short to accommodate a possible value of num=-1" );
```
Comment (JakeChampion) [inline index.js]: I think you have got the arity and use of `Error` confused with that of `console.log`.

The Error constructor function has an arity of 3, all the parameters are optional - `new Error(message, fileName, lineNumber)`

The `console.log` method has an arity of n, all parameters are optional.

To resolve the issue you can concatenate the strings together using `+` operator - `"a" + "b"`. Another solution would be to replace the String literals with a single Template literal ``a ${2+2} ``.

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "share-code-creator",
-  "version": "2.1.2",
-  "description": "",
+  "version": "2.1.3",
```
Comment (JakeChampion) [inline package.json]: It is recommended to not change the version in pull requests.

If the version is not changed in the pull request, the module author can bundle multiple changes together for a single release.

File: index.js [hunk 4]
```diff
@@ -139,9 +164,25 @@ function zeroPadNumToN(num, n) {
 	return zeroPadded;
 }
 
+// For shuffling arrays:
+// - construct a seed from the salt (just use the salt string)
+// - created a seeded prng (via knuth-shuffle-seeded, added to package.json, 
+// -- https://www.npmjs.com/package/knuth-shuffle-seeded)
+// -- https://github.com/TimothyGu/knuth-shuffle-seeded
+// - construct reversible shuffle of article id
```
Comment (JakeChampion) [inline index.js:170]: I don't think these comments are needed, the code itself explains the steps taken

File: index.js (hunk 1 above)
Comment (railsagainstignorance) [inline index.js:246]: so they can be used in the tests

Comment (JakeChampion) [inline index.js:246]: If they are exported, they are part of the public API and will need to be covered by tests.

Comment (railsagainstignorance) [inline index.js:246]: prefixing with _ was the quickest non 100% crap way I could find of exposing the inner functions to allow for better testing whilst hoping that no-one ever uses those functions in the wild.

File: index.js (hunk 2 above)
Comment (railsagainstignorance) [inline index.js]: have updated the Error args

File: package.json (hunk 3 above)
Comment (railsagainstignorance) [inline package.json]: that's me told. ta.

File: index.js (hunk 4 above)
Comment (railsagainstignorance) [inline index.js:170]: beg to differ.
the urls make it easy to look up the details of the seed+shuffle impls.

Comment (JakeChampion) [inline index.js:170]: That can all be found by searching for the package on the registry. We can leave it in.

File: index.js [hunk 5]
```diff
@@ -1,3 +1,7 @@
+'use strict'
```
Comment (JakeChampion) [inline index.js:1]: This is unnecessary inside of a module

File: index.js [hunk 6]
```diff
@@ -1,3 +1,7 @@
...
+
+var shuffle = require('knuth-shuffle-seeded');
+
 var lowercase         = 'abcdefghijklmnopqrstuvwxyz';
```
Comment (JakeChampion) [inline index.js:5]: The multiple spaces is actually against `obt verify` rules

**Pull request merged**